tests/engines/collections/integration/components/collection-submission-confirmation-modal/component-test.ts

```diff
@@ -1,103 +1,183 @@
+        assert.true(externalSaveClicked);
+
     });
 
     test('Cancel button calls cancel action', async function(assert) {
-        assert.expect(1);
-        this.set('noop', () => { /* noop */ });
-        this.set('externalCancelAction', () => {
-            assert.ok(true);
-        });
-        await render(hbs`
-        {{collection-submission-confirmation-modal
-            resubmitToCollection=(action this.noop)
-            openModal=true
-            addToCollection=(action this.noop)
-            cancel=(action this.externalCancelAction)
-        }}`);
+        // Given the component is rendered
+        await render(
+            hbs`<CollectionsSubmission::CollectionSubmissionConfirmationModal
+                @openModal=true
+                @resubmitToCollection={{ this.externalResubmitAction }}
+                @addToCollection={{ this.externalSaveAction}}
+                @cancel={{ this.externalCancelAction}}
+            />`,
+        );
+
+        // Then I assert the cancel button text is correct
+        assert.dom('[data-test-collection-submission-confirmation-modal-cancel-button]')
+            .hasText('Cancel', 'The cancel button is wrong');
+
+        // When I click on the cancel button
+
         await click('[data-test-collection-submission-confirmation-modal-cancel-button]');
+
+        // Then I verify the action was handled
+        assert.true(externalCancelClicked);
     });
 
     test('Resubmit workflow', async function(assert) {
-        this.set('noop', () => { /* noop */ });
-        this.set('resubmit', () => assert.ok(true));
-        await render(hbs`
-        {{collection-submission-confirmation-modal
-            resubmitToCollection=(action this.resubmit)
-            openModal=true 
-            addToCollection=(action this.noop)
-            showResubmitModal=true
-            cancel=(action this.noop)
-        }}`);
+
+        // Given the component is rendered
+        await render(
+            hbs`<CollectionsSubmission::CollectionSubmissionConfirmationModal
+                @resubmitToCollection={{ this.externalResubmitAction }}
+                @showResubmitModal=true
+                @openModal=true
+                @addToCollection={{ this.externalSaveAction}}
+                @cancel={{ this.externalCancelAction }}
+            />`,
+        );
+
+        // Then I verify the modal resubmit text is correct
         assert.dom('[data-test-collection-submission-confirmation-modal-resubmit]').hasText(
-            t('collections.collection_submission_confirmation_modal.body_resubmit'),
+            'This project was removed or rejected from this collection. Do you want to resubmit it?',
+            'The text is valid',
         );
+
+        // And I verify the modal reason text area exists
+        assert.dom('[data-test-collection-submission-confirmation-modal-reason]')
+            .exists('The collection submission confirmation modal reason does not exist.');
```

Comment:
If you use the `sinon.fake` as mentioned above, you can add a comment in this textarea with `await fillIn(selector, comment)` and then verify that the comment is getting passed into the resubmit action.